### PR TITLE
My first set of changes to add support for MPS3 AN547 TSV targ…

### PR DESCRIPTION
**Pull request by mmankal**
…et the command to build is

west build -p always -b mps3_an547_tsv samples/hello_world

when this target is made the image will be generated as zephyr.bin as well as we have turned on generation of zephyr.lst file to ensure we have debug symbols and disassemble code available

The files changes are
Kconfig.zephyr
boards/arm/mps3/board.yml
boards/arm/mps3/mps3_an547_tsv.dts
boards/arm/mps3/mps3_an547_tsv.yaml
boards/arm/mps3/mps3_an547_tsv_defconfig
boards/deprecated.cmake

### CODE REVIEW

**Comment by ithinuel** [comment]
Hello, what does TSV stand for?

File: Kconfig.zephyr [hunk 1]
```diff
@@ -648,12 +648,13 @@ config OUTPUT_SYMBOLS
 
 config OUTPUT_DISASSEMBLY
 	bool "Create a disassembly file"
+	default y
```
Comment (tejlmand) [inline Kconfig.zephyr:651]: this will affect all boards in Zephyr, please do this only for the board (or SoC) where the default should be changed, or provide a better reason why this change should be accepted.

File: Kconfig.zephyr [hunk 2]
```diff
@@ -648,12 +648,13 @@ config OUTPUT_SYMBOLS
...
 	help
 	  Create an .lst file with the assembly listing of the firmware.
 
 config OUTPUT_DISASSEMBLE_ALL
 	bool "Disassemble all sections with source. Fill zeros."
-	default n
+	default y
```
Comment (tejlmand) [inline Kconfig.zephyr:657]: this will affect all boards in Zephyr, please do this only for the board (or SoC) where the default should be changed, or provide a better reason why this change should be accepted.

File: boards/deprecated.cmake [hunk 3]
```diff
@@ -434,7 +434,9 @@ set(mps3_an547_DEPRECATED
 set(mps3_an547_ns_DEPRECATED
     mps3/an547/ns
 )
-
+set(mps3_an547_tsv_DEPRECATED
+    mps3/an547/tsv
+)
```
Comment (tejlmand) [inline boards/deprecated.cmake:439]: this is only needed for boards that are converted from HWMv1 to v2.

`mps3_an547_tsv` doesn't exist as a hwmv1 board.
```suggestion
```

File: boards/arm/mps3/board.yml [hunk 4]
```diff
@@ -5,3 +5,4 @@ board:
   - name: 'an547'
     variants:
     - name: 'ns'
+    - name: 'tsv'
```
Comment (tejlmand) [inline boards/arm/mps3/board.yml:8]: what is a `tsv` board variant ?
Please describe it, what / when is it useful to build for this board.

Remember to update board documentation: https://github.com/zephyrproject-rtos/zephyr/blob/ca38d4bdc0e25cdcf176098454de052dff7b9dbc/boards/arm/mps3/doc/index.rst

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please squash following commits into first commit:
9b1c0e967ddf3de287112803c8872657b88189e0
a2b53c82b4705a6d57890ec2274df08313c58c42
c2f0566df8b290f6e6593b8f00ef1aef73e498d7
fb3409455e3e3dd005afe6cc435af67a017a3c9f

and remove the merge commit: 7fabda416b1b4ae13b1ef74b6c74295045eed934

File: boards/arm/mps3/board.yml (hunk 4 above)
Comment (ithinuel) [reply boards/arm/mps3/board.yml:8]: I asked this question a week ago..

Based on the commiter’s email address, my guess would be that this is a shortened version of `Tsavorite` for `Tsavorite Scalable Intelligence`.

As far as I can see, this new board only differs from the base board on some of the console output and the memory layout/partitions used.
It seems to me this should be a downstream overlay instead of an extra board variant upstream.

File: CMakeLists.txt [hunk 5]
```diff
@@ -2020,6 +2020,41 @@ add_custom_command(
   COMMAND_EXPAND_LISTS
   )
 
+if(CONFIG_BUILD_OUTPUT_TSV_STRIPPED_HEX)
```
Comment (nordicjm) [inline CMakeLists.txt:2023]: No.

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by ithinuel** [comment]
Converting to Draft as:
- the author didn’t respond to previous reviews
- the author seems to continue pushing new changes
- the PR doesn’t pass CI
